utils/crypto/bls/signer/signer.go

```diff
@@ -0,0 +1,114 @@
+// Copyright (C) 2019-2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package signer
+
+import (
+	"context"
+	"encoding/base64"
+	"errors"
+	"fmt"
+	"io/fs"
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/ava-labs/libevm/log"
+
+	"github.com/ava-labs/avalanchego/config/node"
+	"github.com/ava-labs/avalanchego/utils/crypto/bls"
+	"github.com/ava-labs/avalanchego/utils/crypto/bls/signer/localsigner"
+	"github.com/ava-labs/avalanchego/utils/crypto/bls/signer/rpcsigner"
+	"github.com/ava-labs/avalanchego/utils/perms"
+)
+
+var errMissingStakingSigningKeyFile = errors.New("missing staking signing key file")
+
+// GetStakingSigner returns a BLS signer based on the provided configuration.
+func GetStakingSigner(
+	config interface{},
+) (bls.Signer, error) {
+	switch cfg := config.(type) {
+	case node.EphemeralSignerConfig:
+		log.Info("creating ephemeral signer")
+		signer, err := localsigner.New()
+		if err != nil {
+			return nil, fmt.Errorf("couldn't generate ephemeral signer: %w", err)
+		}
+
+		return signer, nil
+
+	case node.ContentKeyConfig:
+		log.Info("creating signer from key content")
+		signerKeyContent, err := base64.StdEncoding.DecodeString(cfg.SignerKeyRawContent)
+		if err != nil {
+			return nil, fmt.Errorf("unable to decode base64 content: %w", err)
+		}
+
+		signer, err := localsigner.FromBytes(signerKeyContent)
+		if err != nil {
+			return nil, fmt.Errorf("couldn't parse signing key: %w", err)
+		}
+
+		return signer, nil
+
+	case node.RPCSignerConfig:
+		log.Info("creating RPC signer")
+		ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+		signer, err := rpcsigner.NewClient(ctx, cfg.StakingSignerRPC)
+		cancel()
+		if err != nil {
+			return nil, fmt.Errorf("couldn't create rpc signer client: %w", err)
+		}
+
+		return signer, nil
+
+	case node.SignerPathConfig:
+		_, err := os.Stat(cfg.SigningKeyPath)
+		if !errors.Is(err, fs.ErrNotExist) {
```

Comment:
Don't we need to handle non-nil errors that aren't of `ErrNotExist` here?

Reply:
So, this was the existing behaviour... What I understand to be the logic here - If we call this function twice with nothing set, we should create a signer from a new key and write that key to a file. If we call this function again, it should use the file that was written instead of creating a second new key.

If there's an error that isn't `fs.ErrNotExist`, it will error again in `createSignerFromFile` and bubble that error up anyway.